pkg/controller/cluster/custom.go

```diff
@@ -0,0 +1,146 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2021 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math"
+	"time"
+
+	miniov2 "github.com/minio/operator/pkg/apis/minio.min.io/v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var secretTypePublicKeyNameMap = map[string]string{
+	"kubernetes.io/tls":        "tls.crt",
+	"cert-manager.io/v1":       "tls.crt",
+	"cert-manager.io/v1alpha2": "tls.crt",
+	// Add newer secretTypes and their corresponding values in future
+}
+
+// getCustomCertificates
+func (c *Controller) getCustomCertificates(ctx context.Context, tenant *miniov2.Tenant) (customCertificates *miniov2.CustomCertificates, err error) {
+	namespace := tenant.Namespace
+	secretsMap := map[string][]*miniov2.LocalCertificateReference{
+		"Client":   tenant.Spec.ExternalCertSecret,
+		"Minio":    tenant.Spec.ExternalClientCertSecrets,
```

Comment:
```suggestion
		"Client":   tenant.Spec.ExternalClientCertSecrets,
		"Minio":    tenant.Spec.ExternalCertSecret,
```
I think `tenant.Spec.ExternalClientCertSecrets` should be mapped to `Client` ?

Reply:
nice catch!

Reply:
Resolved. Thanks!